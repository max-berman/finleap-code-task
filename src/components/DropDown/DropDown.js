import React from 'react'
import { CITIES } from '../../cities'
import fetchCity from '../../api'
import { uniqueBy } from '../../utils'

export default function List({
  userInput,
  setUserInput,
  setCityList,
  setShowDropDown,
  showDropDown,
  setIsError,
  setIsLoading
}) {
  function handleItemClick(id) {
    setIsLoading(true)
    return fetchCity(id)
      .then(({ id, main: { temp_max, temp_min }, name }) => {
        setCityList(prevState =>
          uniqueBy([
            ...prevState,
            {
              id,
              name,
              temp_max,
              temp_min
            }
          ])
        )
        setUserInput(name)
      })
      .then(() => {
        setShowDropDown(!showDropDown)
        setIsLoading(false)
      })
      .catch(err => setIsError(true))
  }

  return showDropDown ? (
    <ul className='dropDownList' data-testid='dropDown'>
      {CITIES.filter(
        res => res.name.toLowerCase().indexOf(userInput.toLowerCase()) > -1
      ).map(({ name, id }, i) => (
        <li key={i} onClick={() => handleItemClick(id)}>
          {name}
        </li>
      ))}
    </ul>
  ) : null
}
