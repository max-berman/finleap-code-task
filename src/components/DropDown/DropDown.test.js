import React from 'react'
import { render, cleanup } from '@testing-library/react'
import DropDown from './DropDown'

afterEach(cleanup)

describe('<DropDown /> spec', () => {
  it('renders the component', () => {
    const container = render(<DropDown />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
