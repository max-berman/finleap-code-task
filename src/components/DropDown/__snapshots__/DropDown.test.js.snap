// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropDown /> spec renders the component 1`] = `undefined`;
